pandas/core/reshape/merge.py

```diff
@@ -1200,7 +1209,34 @@ def _maybe_coerce_merge_keys(self):
                 typ = rk.categories.dtype if rk_is_cat else object
                 self.right = self.right.assign(**{name: self.right[name].astype(typ)})
 
+    def _create_cross_configuration(
+        self, _left, _right, on
+    ) -> Tuple["DataFrame", "DataFrame", str, str]:
+        if on is not None:
+            raise MergeError(
+                "Can not pass any merge columns when using cross as merge method"
+            )
+        cross_col = f"{max([*_left.columns, *_right.columns])}_cross"
+        _left = _left.copy()
+        _right = _right.copy()
+        _left.insert(loc=0, value=1, column=cross_col)
```

Comment:
you can just use .assign (IOW put it at the end)

Reply:
Done